### PR TITLE
Use Infer to detect image/video

### PR DESCRIPTION
**Pull request by Pyroglyph**
Maintaining our own list of known filetypes seems pointless when there are other projects like [Infer](https://crates.io/crates/infer) that do it for us.

Marked as draft until this is answered:
If Infer fails to load the file then `ProbeError`s are created despite not being thrown by FFProbe. This is because the check happens in the `probe` function (for convenience). Is this okay or should a more specific error type be created for this? In any case, the whole thing would have failed anyway if the file is unable to be read.

### CODE REVIEW

**Comment by alexheretic** [comment]
Thanks! The additional binary bloat and sync-io blocking is undesirable, but overall this looks worth it.

**Comment by alexheretic** [comment]
> If Infer fails to load the file then ProbeErrors are created despite not being thrown by FFProbe. This is because the check happens in the probe function (for convenience). Is this okay or should a more specific error type be created for this? In any case, the whole thing would have failed anyway if the file is unable to be read.

Yeah it's kinda academic as we'd expect the probe to have failed. I probably would have just `Err(_) => false` to reduce the code, but returning an error as you have done is fine too. I don't think we should throw more code at this scenario though.

**Comment by Pyroglyph** [comment]
> The additional binary bloat is undesirable
Infer can be imported with no default features to make it lighter. This means that I won't be able to use `infer::get_from_path` helper any more, but if size is something we're concerned about then it should just be a matter of changing a few lines.

**Comment by alexheretic** [comment]
Size is a concern, if not a critical one. Just something I keep an eye on when adding new dependencies. As I said I think the trade off is worth it here.

**Comment by Pyroglyph** [comment]
I've done some measurements. When building in `--release` mode I get these file sizes:

Version|File size
-|-
No Infer|2,363 KB
Infer (default features)|2,445 KB (+82 KB)
Infer (no default features)|2,374 KB (+11 KB)

I'd say that's quite a nice jump.

File: src/ffprobe.rs [hunk 1]
```diff
@@ -85,11 +78,24 @@ pub fn probe(input: &Path) -> Ffprobe {
...
 
+fn is_image(path: &Path) -> anyhow::Result<bool> {
+    // This code was taken from infer::get_from_path, but it's not available since we're importing it as no_std
+    let file = File::open(path)?;
+    let limit = file
+        .metadata()
+        .map(|m| std::cmp::min(m.len(), 8192) as usize + 1)
+        .unwrap_or(0);
+    let mut file_header = Vec::with_capacity(limit);
+    file.take(8192).read_to_end(&mut file_header)?;
```
Comment (alexheretic) [inline src/ffprobe.rs:94]: I know you've taken this code from _infer_ but it is a bit odd.

I imagine querying metadata will be slower than any micro gain from allocating a smaller number than 8K, which we don't really expect anyway.

```suggestion
    let file = File::open(path)?;
    let mut file_header = Vec::with_capacity(8192);
    file.take(8192).read_to_end(&mut file_header)?;
```

**Comment by alexheretic** [comment]
It's a mild annoyance that we don't need no_std, we just want to avoid the _cfb_ dependency. But _infer_ has bundled them together in the "std" feature :shrug:.

**Review by alexheretic — APPROVED** [review]
lgtm

**Pull request merged**